test/unit/repositories/test_manufacturer.py

```diff
@@ -16,588 +22,652 @@
+
+        # Pass through ManufacturerIn first as need creation and modified times
+        self._manufacturer_in = ManufacturerIn(**manufacturer_in_data)
+
+        self._expected_manufacturer_out = ManufacturerOut(
+            **self._manufacturer_in.model_dump(), id=inserted_manufacturer_id
+        )
+        #
+        self.mock_is_duplicate_manufacturer(duplicate_manufacturer_in_data)
+        # Mock 'insert one' to return object for inserted manufacturer
+        RepositoryTestHelpers.mock_insert_one(self.manufacturers_collection, inserted_manufacturer_id)
+        # Mock 'find_one' to return the inserted manufacturer document
+        RepositoryTestHelpers.mock_find_one(
+            self.manufacturers_collection, {**self._manufacturer_in.model_dump(), "_id": inserted_manufacturer_id}
+        )
+
+    def call_create(self) -> None:
+        """
+        Calls the `ManufacturerRepo` `create` method with the appropriate data from a prior call to `mock_create`.
+        """
+        self._created_manufacturer = self.manufacturer_repository.create(
+            self._manufacturer_in, session=self.mock_session
+        )
+
+    def call_create_expecting_error(self, error_type: type[BaseException]) -> None:
+        """
+        Calls the `ManufacturerRepo` `create` method with the appropriate data from a prior call to `mock_create` while
+        expecting an error to be raised.
+
+        :param error_type: Expected exception to be raised.
+        """
+        with pytest.raises(error_type) as exc:
+            self.manufacturer_repository.create(self._manufacturer_in, session=self.mock_session)
+        self._create_exception = exc
+
+    def check_create_success(self) -> None:
+        """
+        Checks that a prior call to `call_create` worked as expected.
+        """
+        manufacturer_in_data = self._manufacturer_in.model_dump()
+
+        # Obtain a list of expected find_one calls
+        expected_find_one_calls = [
+            # This is the check for the duplicate
+            self.get_is_duplicate_manufacturer_expected_find_one_call(self._manufacturer_in, None),
+            # This is the check for the newly inserted manufacturer get
+            call({"_id": CustomObjectId(self._expected_manufacturer_out.id)}, session=self.mock_session),
         ]
-    )
-    assert created_manufacturer == manufacturer_out
+        self.manufacturers_collection.find_one.assert_has_calls(expected_find_one_calls)
+
+        self.manufacturers_collection.insert_one.assert_called_once_with(
+            manufacturer_in_data, session=self.mock_session
+        )
+        assert self._created_manufacturer == self._expected_manufacturer_out
+
+    def check_create_failed_with_exception(self, message: str) -> None:
+        """
+        Checks that a prior call to `call_create_expecting_error` worked as expected, raising an exception with the
+        correct message.
 
+        :param message: Message of the raised exception.
```

Comment:
```suggestion
        :param message: Expected message of the raised exception.
```
Same for any others, I think this is clearer.